synapse/handlers/federation.py

```diff
@@ -796,9 +796,15 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase):
             return
 
         # Skip processing a marker event if the room version doesn't
-        # support it.
+        # support it or the event is not from the room creator.
         room_version = await self.store.get_room_version(marker_event.room_id)
-        if not room_version.msc2716_historical:
+        create_event = await self.store.get_create_event_for_room(marker_event.room_id)
+        room_creator = create_event.content.get("creator", None)
+        if (
+            not room_version.msc2716_historical
+            or not self.hs.config.experimental.msc2716_enabled
```

Comment:
> Now that we're adding code that's not behind an experimental room version can you add an experimental config option please? This is mainly so that it's easy to find all code related to an experimental MSC
>
> https://github.com/matrix-org/synapse/pull/10566#issuecomment-906437909

Added ⏩